spec/support/controller_helpers.rb

```diff
@@ -0,0 +1,19 @@
+RSpec.shared_examples 'an authenticated resource action' do |valid_header|
+  if valid_header
```

Comment:
you don't use if in tests, create an shared example for the authenticated and not authenticated case